dipy/reconst/mapmri.py

```diff
@@ -261,97 +462,490 @@ def odf(self, sphere, s=0):
+        NeuroImage 2015, in press.
+        """
+        if self.model.bval_threshold > 2000.:
+            msg = 'model bval_threshold must be lower than 2000 for the '
+            msg += 'non_Gaussianity to be physically meaningful [2].'
+            warn(msg)
+        if not self.model.anisotropic_scaling:
+            msg = 'Parallel non-Gaussianity is not defined using '
+            msg += 'isotropic scaling.'
+            raise ValueError(msg)
+
+        ind_mat = self.model.ind_mat
+        coef = self._mapmri_coef
+        a_perp = np.zeros_like(coef)
+        a00 = np.zeros_like(coef)
+
+        for i in range(coef.shape[0]):
+            n1, n2, n3 = ind_mat[i]
+            if n1 % 2 == 0:
+                if n2 % 2 == 0 and n3 % 2 == 0:
+                    a_perp[i] = coef[i] * (-1) ** (n1 / 2) *\
+                        np.sqrt(factorial(n1)) / factorial2(n1)
+                    if n2 == 0 and n3 == 0:
+                        a00[i] = a_perp[i]
+        return np.sqrt(1 - np.sum(a00 ** 2) / np.sum(a_perp ** 2))
+
+    def norm_of_laplacian_signal(self):
+        """ Calculates the norm of the laplacian of the fitted signal [1]_.
+        This information could be useful to assess if the extrapolation of the
+        fitted signal contains spurious oscillations. A high laplacian may
+        indicate that these are present, and any q-space indices that
+        use integrals of the signal may be corrupted (e.g. RTOP, RTAP, RTPP,
+        QIV).
+
+        References
+        ----------
+        .. [1]_ Fick, Rutger HJ, et al. "MAPL: Tissue microstructure estimation
+        using Laplacian-regularized MAP-MRI and its application to HCP data."
+        NeuroImage (2016).
+        """
+        if self.model.anisotropic_scaling:
+            laplacian_matrix = mapmri_laplacian_reg_matrix(
+                    self.model.ind_mat, self.mu,
+                    self.model.S_mat, self.model.T_mat, self.model.U_mat)
         else:
-            tau = gtab.big_delta - gtab.small_delta / 3.0
+            laplacian_matrix = self.mu[0] * self.model.laplacian_matrix
 
-        qvals = np.sqrt(gtab.bvals / tau) / (2 * np.pi)
-        qvecs = np.dot(gtab.bvecs, self.R)
-        q = qvecs * qvals[:, None]
-        s_mat = mapmri_phi_matrix(self.radial_order, self.mu, q.T)
-        S_reconst = S0 * np.dot(s_mat, self._mapmri_coef)
+        norm_of_laplacian = np.dot(np.dot(self._mapmri_coef, laplacian_matrix),
+                                   self._mapmri_coef)
+        return norm_of_laplacian
 
-        return S_reconst
+    def fitted_signal(self, gtab=None):
+        """
+        Recovers the fitted signal for the given gradient table. If no gradient
+        table is given it recovers the signal for the gtab of the data.
+        """
+        if gtab is None:
+            E = self.predict(self.model.gtab)
+        else:
+            E = self.predict(gtab)
+        return E
+
+    def predict(self, qvals_or_gtab, S0=1.):
```

Comment:
We're trying to be more uniform about this, setting S0 per default to 100 (https://github.com/nipy/dipy/pull/1063)